extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -385,22 +385,22 @@ class LifecycleTask extends BackbeatTask {
             if (rule.Status === 'Disabled') {
                 return false;
             }
-            if (rule.Filter && rule.Filter.And) {
-                // multiple tags or prefix w/ tag(s)
-                const prefix = rule.Filter.And.Prefix;
-                if (prefix && !item.Key.startsWith(prefix)) {
-                    return false;
-                }
-                return deepCompare(rule.Filter.And.Tags, objTags.TagSet);
-            }
-            // only one prefix or one tag to compare in this rule
+            // check all locations where prefix could possibly be
             const prefix = rule.Prefix ||
-                (rule.Filter && rule.Filter.Prefix);
+                  (rule.Filter && (rule.Filter.And ?
```

Comment:
Isn't the second part of the conditional returning a boolean and not the actual prefix?

Reply:
If you talk specifically about `rule.Filter.And ? .. : ..`, it's the ternary operator that returns either the 2nd or 3rd operand depending on the evaluation of the 1st, so not a boolean.

Reply:
Thanks the `&&` throw me off